### PR TITLE
feat(types): model.get overloading reordering

### PR DESCRIPTION
**Pull request by mirkojotic**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change
Solves #10782 

Adding a test and reordering overloading

### CODE REVIEW

File: types/test/model.ts [hunk 1]
```diff
@@ -10,3 +11,6 @@ class MyModel extends Model {
 }
 
 const assoc: Association = MyModel.associations.other;
+
+const Instance: MyModel = new MyModel({ int: 10 });
+const int: IntegerDataType = Instance.get('int');
```
Comment (SimonSchick) [inline types/test/model.ts]: `IntegerDataType` is not used for model properties but only inside of sequelize, you should use something like `number`.

Comment (mirkojotic) [reply types/test/model.ts]: Should be good now.

**Comment by sushantdhiman** [comment]
@mirkojotic Rebase required

**Pull request merged**